### ISSUE TITLE
Support tilemaps with no texture / image asset (tiles are flat color)

### ISSUE THREAD

**Issue by inodentry** [StarArawn/bevy_ecs_tilemap#370]
I'd like to create a tilemap editor application, and so I want the ability to make a tilemap to use as an overlay for showing selections/highlights/cursors/etc. There is no need for a texture, only flat color tiles.

Currently, there is no way to do this. I would need to create some simple image asset with just a flat color tile. This is a lot of effort, especially since it requires the tile/pixel dimensions to match.

It would be nice if I didn't have to do this.

Perhaps add a `TilemapTexture::None` variant? It could be implemented so that it skips all the atlas/array code in `bevy_ecs_tilemap`. If the shader needs a texture binding, you could use the default fallback texture (1x1 white pixel) provided by `bevy_render`. Alternatively, use pipeline specialization and shader defs to skip texture sampling in the shader entirely.

**Comment by Andrew-Pynch** [comment]
Incorrect, as demonstrated in examples/colors.rs this is easily possible. 
https://github.com/StarArawn/bevy_ecs_tilemap/blob/main/examples/colors.rs

For example, if you wanted to create a tile that was green you could do something like 
```rs
    fill_tilemap_rect_color(
        TileTextureIndex(5),
        TilePos {
            x: QUADRANT_SIDE_LENGTH,
            y: 0,
        },
        quadrant_size,
        Color::rgba(0.0, 1.0, 0.0, 1.0),
        tilemap_id,
        &mut commands,
        &mut tile_storage,
    );
    ```

Hope this helps

**Comment by StarArawn** [comment]
> Incorrect, as demonstrated in examples/colors.rs this is easily possible. 

This isn't accurate. Although it is possible to create a tilemap and color tiles using a Color. It's not possible to create a renderable tilemap without a texture see:
https://github.com/StarArawn/bevy_ecs_tilemap/blob/main/examples/colors.rs#L85

**Comment by inodentry** [comment]
Yes, this is about being able to easily spawn tilemaps without needing any texture/image asset. I know colors can be set per-tile.
